#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 21c6c0898a1de1355d6731ec228215ce66b27f2c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ingo=20Kl=C3=B6cker?= <dev@ingo-kloecker.de>
Date: Mon, 4 Oct 2021 11:01:16 +0200
Subject: [PATCH] Ask GnuPG for the correct path for the uiserver socket

This ensures that clients find the uiserver socket used by Kleopatra.

GnuPG-bug-id: 5619
---
 src/libkleopatraclient/core/command.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/libkleopatraclient/core/command.cpp b/src/libkleopatraclient/core/command.cpp
index 443165cb..88d1929a 100644
--- a/src/libkleopatraclient/core/command.cpp
+++ b/src/libkleopatraclient/core/command.cpp
@@ -433,6 +433,11 @@ static QString gnupg_home_directory()
 
 static QString get_default_socket_name()
 {
+    const QString socketPath{QString::fromUtf8(GpgME::dirInfo("uiserver-socket"))};
+    if (!socketPath.isEmpty()) {
+        // Note: The socket directory exists after GpgME::dirInfo() has been called.
+        return socketPath;
+    }
     const QString homeDir = gnupg_home_directory();
     if (homeDir.isEmpty()) {
         return QString();
-- 
2.33.0

